src/main/java/com/github/checkstyle/regression/git/DiffParser.java

```diff
@@ -93,32 +90,27 @@ private DiffParser() {
     }
 
     /**
-     * Creates a tree parser from a commit, to be used by diff command.
-     * @param repository the repository to parse diff
-     * @param ref        the name of the ref to the commit; e.g., "refs/heads/master"
-     * @return the tree parser
+     * Gets a TreeParserPair for the further use.
+     * @param repository the repository to parse
+     * @param branch     the name of the feature branch
+     * @return the TreeParserPair prepared for the further use
      * @throws IOException JGit library exception
      */
-    private static AbstractTreeIterator prepareTreeParser(Repository repository, String ref)
+    private static TreeParserPair getTreeParserPair(Repository repository, String branch)
             throws IOException {
-        final CanonicalTreeParser returnValue;
+        final TreeParserPair returnValue = new TreeParserPair();
         final RevWalk walk = new RevWalk(repository);
 
         try {
-            final Ref head = repository.exactRef(ref);
-            final RevCommit commit = walk.parseCommit(head.getObjectId());
-            final RevTree tree = walk.parseTree(commit.getTree().getId());
-            final ObjectReader reader = repository.newObjectReader();
-
-            try {
-                returnValue = new CanonicalTreeParser();
-                returnValue.reset(reader, tree.getId());
-            }
-            finally {
-                reader.close();
-            }
-
+            final RevCommit featureCommit = walk.parseCommit(
+                    repository.exactRef(Constants.R_HEADS + branch).getObjectId());
+            final RevCommit masterCommit = walk.parseCommit(
+                    repository.exactRef(Constants.R_HEADS + "master").getObjectId());
+            final RevCommit baseCommit = getBaseCommit(walk, featureCommit, masterCommit);
             walk.dispose();
+
+            returnValue.featureTreeParser = prepareTreeParser(walk, featureCommit);
+            returnValue.baseTreeParser = prepareTreeParser(walk, baseCommit);
```

Comment:
Shoudn't `TreeParserPair` have a constructor that takes these 2 as an argument to make sure they are always set together.